File: Examples/test-suite/python/inherit_missing_runme.py
import inherit_missing

a = inherit_missing.new_Foo()
b = inherit_missing.Bar()
c = inherit_missing.Spam()

x = inherit_missing.do_blah(a)
if x != "Foo::blah":
    raise RuntimeError("Whoa! Bad return {}".format(x))

x = inherit_missing.do_blah(b)
if x != "Bar::blah":
    raise RuntimeError("Whoa! Bad return {}".format(x))

x = inherit_missing.do_blah(c)
if x != "Spam::blah":
    raise RuntimeError("Whoa! Bad return {}".format(x))

inherit_missing.delete_Foo(a)

File: Examples/test-suite/javascript/inherit_missing_runme.js
var inherit_missing = require("inherit_missing");

a = inherit_missing.new_Foo()
b = new inherit_missing.Bar()
c = new inherit_missing.Spam()

x = inherit_missing.do_blah(a)
if (x != "Foo::blah")
    throw new Error("Whoa! Bad return {}".format(x))

x = inherit_missing.do_blah(b)
if (x != "Bar::blah")
    throw new Error("Whoa! Bad return {}".format(x))

x = inherit_missing.do_blah(c)
if (x != "Spam::blah")
    throw new Error("Whoa! Bad return {}".format(x))

inherit_missing.delete_Foo(a)
